From 0e8982eefcec2ea9dde0451423505f17b86eaf0a Mon Sep 17 00:00:00 2001
From: Henrik Grimler <henrik@grimler.se>
Date: Sat, 27 Feb 2021 17:56:32 +0100
Subject: [PATCH] security: tima_uevent: fix build error due to out-of-src
 build or gcc version

Not sure exactly what triggers the error, but properly setting up
Makefiles and Kconfig for tima_uevent fixes it. Previously compilation
failed with:

security/tima_uevent/tima_uevent.c:217:1: fatal error: opening dependency file security/tima_uevent/.tima_uevent.o.d: No such file or directory
  217 | module_exit(tima_uevent_exit);
      | ^~~~~~~~~~~

This commit is basically the same as (part of)
https://github.com/jcadduono/android_kernel_samsung_universal3475/commit/8b34f9d012ccf836f6211fc7577d40f460402718

Change-Id: I8a7625c820df5e19b44567df12275cc90b8c862a
---
 security/Kconfig              |  2 +-
 security/Makefile             |  2 +-
 security/tima_uevent/Kconfig  | 17 +++++++----------
 security/tima_uevent/Makefile |  3 +--
 4 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/security/Kconfig b/security/Kconfig
index 0351ba7e4ed..e332c9225bb 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -133,6 +133,7 @@ source security/apparmor/Kconfig
 source security/yama/Kconfig
 
 source security/integrity/Kconfig
+source security/tima_uevent/Kconfig
 source security/tz_iccc/Kconfig
 
 choice
@@ -178,4 +179,3 @@ config DEFAULT_SECURITY
 	default "" if DEFAULT_SECURITY_DAC
 
 endmenu
-
diff --git a/security/Makefile b/security/Makefile
index 36d3f0d176b..597e4f0369a 100644
--- a/security/Makefile
+++ b/security/Makefile
@@ -30,7 +30,7 @@ subdir-$(CONFIG_INTEGRITY)		+= integrity
 obj-$(CONFIG_INTEGRITY)			+= integrity/built-in.o
 
 # TIMA uevent 
-obj-$(CONFIG_TIMA)     += tima_uevent/tima_uevent.o
+obj-$(CONFIG_TIMA)     += tima_uevent/
 
 #TZ ICCC
 obj-$(CONFIG_TZ_ICCC)	  += tz_iccc/
diff --git a/security/tima_uevent/Kconfig b/security/tima_uevent/Kconfig
index fd7ea958b75..5c8006f19b3 100644
--- a/security/tima_uevent/Kconfig
+++ b/security/tima_uevent/Kconfig
@@ -2,14 +2,11 @@
 # TIMA uevent configuration
 #
 
-menuconfig TIMA_UEVENT_MENU
-    tristate "TIMA uevent support"
-    help
-      TIMA uevent device and driver provides a mechanism to propogate
-      periodic kernel and module measurement events to observer. 
-
 config TIMA_UEVENT
-    bool "TIMA uevents (EXPERIMENTAL)"
-    depends on EXPERIMENTAL
-    ---help---
-    Generate udev events for TIMA events.
+	bool "TIMA uevents (EXPERIMENTAL)"
+	depends on EXPERIMENTAL
+	  default n
+	---help---
+	  Generate udev events for TIMA events.
+	  TIMA uevent device and driver provides a mechanism to propogate
+	  periodic kernel and module measurement events to observer.
\ No newline at end of file
diff --git a/security/tima_uevent/Makefile b/security/tima_uevent/Makefile
index 4f1ca82645b..ae0c213e371 100644
--- a/security/tima_uevent/Makefile
+++ b/security/tima_uevent/Makefile
@@ -2,5 +2,4 @@
 # Makefile for TIMA kernel uevent drivers.
 #
 
-#obj-$(CONFIG_TIMA)   += tima_uevent.o
-
+obj-$(CONFIG_TIMA) += tima_uevent.o
